### ISSUE TITLE
Question regarding the "Fallback option" in compute_bboxes_area_loss and its role in the methodological framework

### ISSUE THREAD

**Issue by ShuchunXu** [purseclab/DetStorm#6]
Dear Authors,
While performing a detailed code review and reproduction of DETSTORM for my research, I noticed a specific logic in the `compute_bboxes_area_loss` function that seems to be a supplement to the methodology described in the paper.
**Observations**: When the number of detected boxes is not greater than one, the code[BatchReferenceImgs_attack.py](https://github.com/purseclab/DetStorm/blob/main/DetStorm/BatchReferenceImgs_attack.py) switches from calculating distances/intersections to a "Fallback option",However, this part is not explained in the paper.:
```
else:
    # Fallback option: calculate bboxes' area avg
    bboxes_x1_wh = xyxy2xywh(boxes_x1)[:, 2:]
    bboxes_x1_area = bboxes_x1_wh[:, 0] * bboxes_x1_wh[:, 1]
    img_loss = bboxes_x1_area.mean() / (patch_size[0] * patch_size[1])
```

My Questions:
1. Purpose: What is the specific motivation behind penalizing the average area of bounding boxes when only a single box is detected? Is this intended to prevent the optimizer from converging on a single large "global" bounding box that covers the entire patch?
2. Role in the Framework: How does this fallback mechanism interact with the overall goal of maximizing NMS latency? Does it serve as a "warm-up" phase to guide the optimization from a single-object detection state to a multi-object state, where the more complex distance/intersection losses can take over?
3. Experimental Impact: In your experiments, did you find this fallback to be essential for avoiding local minima or optimization stagnation during the early stages of perturbation generation?

I would appreciate any insights into the engineering rationale behind this choice, as it provides a fascinating look into the practical realization of the theoretical framework.

Best regards.

**Comment by webcrawlr** [comment]
1.  You are correct in the purpose of the loss function itself; it's intended to prevent it from converging on a large, all-covering bounding box, and tries to make the box "smaller" to attempt to eventually generate more boxes around it.
2. It wasn't intended to be a warmup, but it might unintentionally serve that role.  I mainly stuck it there to prevent there from being errors with one bounding box (since distance calculations don't make sense with only one bounding box).  I had to come up with something, and minimizing based on area average seemed like a reasonable approach.
3. It was an edge case we didn't encounter super often, and I didn't spend a lot of time examining single bounding box scenarios, but thinking about it theoretically, it's definitely possible that if the system got stuck on a single bounding box, the fallback option could prep the system to generate more per surface by giving it more area to work with.  I don't know for sure though.

**Comment by ShuchunXu** [comment]
> 1. You are correct in the purpose of the loss function itself; it's intended to prevent it from converging on a large, all-covering bounding box, and tries to make the box "smaller" to attempt to eventually generate more boxes around it.
> 2. It wasn't intended to be a warmup, but it might unintentionally serve that role.  I mainly stuck it there to prevent there from being errors with one bounding box (since distance calculations don't make sense with only one bounding box).  I had to come up with something, and minimizing based on area average seemed like a reasonable approach.
> 3. It was an edge case we didn't encounter super often, and I didn't spend a lot of time examining single bounding box scenarios, but thinking about it theoretically, it's definitely possible that if the system got stuck on a single bounding box, the fallback option could prep the system to generate more per surface by giving it more area to work with.  I don't know for sure though.

Dear Author,
Thank you for the detailed clarification! It makes perfect sense that the "fallback area loss" acts as a catalyst for object splitting.
I have a follow-up question regarding your **initial validation phase.** Since generating the full perturbation dictionary for 119 classes on 10k images takes weeks, how did you quickly verify that the DETSTORM idea was viable during the early stages of your research?
Specifically, I am curious about your pilot experiment setup:
1. **Mini-Dictionary Strategy**: Did you start by generating a "Mini-Dictionary" targeting only a few high-priority classes (e.g., Car, Pedestrian, Truck) on a much smaller subset of images (e.g., 100-500 images)?
2. **Scene Selection for Prototype**: When you were first testing if "Zone Stitching" could actually trigger NMS latency in real-time, how did you pick your initial test scenes? Did you focus on "object-dense" urban scenarios to maximize the potential for latency early on?
3. **Optimization Trade-offs**: In the early proof-of-concept phase, did you use fewer PGD iterations or a lower resolution to speed up the feedback loop?

Understanding how you pivoted from the initial idea to a verifiable prototype without the full-scale computational burden would be extremely helpful for researchers working with limited compute resources.
Best regards.

**Comment by webcrawlr** [comment]
For the pilot experiments, we didn't use stitching at all.  Instead, we generated adversarial examples against the entire image (similar to Phantom Sponges).  We did those over maybe a dozen images.  This approach had issues (some of which we highlight in the paper, such as certain zones having more bounding boxes than others), but it was the most effective test to see if we could generate latency with the new loss functions.  After that, we went straight into doing the full perturbation dictionary.

I believe for resource-constrained experiments, it would be best to find BDD100K or other image examples where you have as much of the screen covered by surface as possible, with ideally not too many classes, and then generate specific adversarial examples on the entire image.  This will provide as close a surrogate to full DetStorm as possible while taking much less time.
